azurerm/internal/services/redis/resource_arm_redis_cache.go

```diff
@@ -672,6 +672,12 @@ func expandRedisConfiguration(d *schema.ResourceData) (map[string]*string, error
 
 	// RDB Backup
 	if v, ok := d.GetOk("redis_configuration.0.rdb_backup_enabled"); ok {
+		backupEnabled := v.(bool)
+		if backupEnabled == true {
+			if _, ok := d.GetOk("redis_configuration.0.rdb_storage_connection_string"); !ok {
```

Comment:
we should also check that its empty
```suggestion
			if v, ok := d.GetOk("redis_configuration.0.rdb_storage_connection_string"); !ok || v.(string) != ""{
```